OutputResults/OutputResults.c

```diff
@@ -0,0 +1,52 @@
+#include "OutputResults.h"
+
+int outputPoints(char* filename) {
+    FILE* fp;
+    struct point* point;
+
+    // check if there are points to print
+    if (inputShape->numOfPoints <= 0) {
+        #ifdef DEBUG
+            printf("ERROR no points to print\n");
+        #endif
+        return 0;
+    }
+
+    // check for valid filename
+    if (filename == NULL) {
+        #ifdef DEBUG
+            printf("ERROR null filename\n");
+        #endif
+        return 0;
+    }
+    
+    // open file in write mode
+    if ((fp = fopen(filename, "w")) == NULL) {
+        #ifdef DEBUG
+            printf("ERROR opening file\n");
+        #endif
+        return 0;
+    }
+
+    // output points to file
+    for (int i = 0; i < inputShape->numOfPoints; i++) {
```

Comment:
Can remove this error easily that you have by directly using the getPoint call untill false incrementing in a while loop. This removes dependency on uninitiated values and retains functionality of code.

Reply:
After implementing this change the error moves to the getPoint function since it also uses inputShape->numOfPoints (see error below).
A temporary solution is to set numOfPoints in the readInput function, however, this may cause a merge conflict for group 1. 
![image](https://user-images.githubusercontent.com/75465982/141860943-f7aa484f-486b-4eb8-b5cf-eb072bf6419b.png)